src/publish.proj

```diff
@@ -6,14 +6,30 @@
   <Import Project="$(ToolsDir)PublishContent.targets" />
   <Import Project="$(ToolsDir)versioning.targets" />
 
+  <!-- gathers the items to be published -->
+  <Target Name="GatherItemsForPattern">
+    <Error Condition="'$(PublishPattern)' == ''" Text="Please specify a value for PublishPattern using standard msbuild 'include' syntax." />
+    <ItemGroup>
+      <ForPublishing Include="$(PublishPattern)" />
+    </ItemGroup>
+    <!-- add relative blob path metadata -->
+    <ItemGroup>
+      <ForPublishing>
+        <RelativeBlobPath>$(BlobNamePrefix)$([System.String]::Copy('%(RecursiveDir)%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath>  
```

Comment:
Hmm, I can imagine also doing this without overriding this target by changing the pattern to (effectively):
`$(PackagesDir)AzureTransfer\**\*.nupkg`
then filter down to the ones in `Release` (using `Exclude`?), so that `RecursiveDir` ends up correct. I'm not sure if it's less complex than what you've done here though.

Reply:
I'm missing something.  Why does this target need to be duplicated?  This looks the same as what is in buildtools except it doesn't add a trailing slash to BlobNamePrefix.

Reply:
That wasn't added to buildtools until 1.1.0, 1.0.0 just sets `RelativeBlobPath` to the local relative path: https://github.com/dotnet/buildtools/blob/release/1.0.0/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/PublishContent.targets#L19

Reply:
The difference is the BlobNamePrefix vs. https://github.com/dotnet/buildtools/blob/release/1.0.0/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/PublishContent.targets#L19

```xml
<RelativeBlobPath>$([System.String]::Copy('%(RecursiveDir)%(Filename)%(Extension)').Replace('\' ,'/'))</RelativeBlobPath>
```

Reply:
Ah, I was looking at master, thanks for pointing that out.

Reply:
@dagood I did it this way because we already have this pattern in CoreClr: https://github.com/dotnet/coreclr/blob/release/1.0.0/src/publish.proj#L9-L21

Reply:
Ah, cool, sounds good to me then.